packages/docsify-server-renderer/index.js

```diff
@@ -1,18 +1,45 @@
 import { readFileSync } from 'fs';
 import { resolve, basename } from 'path';
-import resolvePathname from 'resolve-pathname';
 import { AbstractHistory } from '../../src/core/router/history/abstract';
 import { Compiler } from '../../src/core/render/compiler';
 import { isAbsolutePath } from '../../src/core/router/util';
 import * as tpl from '../../src/core/render/tpl';
 import { prerenderEmbed } from '../../src/core/render/embed';
+import resolvePathname from 'resolve-pathname';
 import fetch from 'node-fetch';
 import debug from 'debug';
+import DOMPurify from 'dompurify';
 
 function cwd(...args) {
   return resolve(process.cwd(), ...args);
 }
 
+function isExternal(url) {
+  let match = url.match(
+    /^([^:\/?#]+:)?(?:\/\/([^\/?#]*))?([^?#]+)?(\?[^#]*)?(#.*)?/
```

Comment:
We can probably remove some of the escape chars here. I like to test regexes over at https://regexr.com. Besides the nice regex features, it has a nice account system: you can log in and save your favorite regexes, and also see others from the community.

Reply:
I think using this approach may be easier: https://j11y.io/snippets/getting-a-fully-qualified-url/

I'll make a PR onto this PR...